apps/android/boot.js

```diff
@@ -230,6 +232,42 @@
         }
         gbSend({t: "actfetch", state: "end", count: actCount});
       },
+      //{t:"listRecs", id:"20230616a"}
+      "listRecs": function() {
+        let recs = require("Storage").list(/^recorder\.log.*\.csv$/,{sf:true}).map(s => s.slice(12, 21));
+        if (event.id.length > 2) { // Handle if there was no id supplied. Then we send a list all available recorder logs back.
+        recs = recs.slice(recs.indexOf(event.id)); // FIXME: Currently does not handle if the last fetched log was deleted on the watch, so that there won't be any entry corresponding to `event.id`
+        recs.shift();
+        }
+        gbSend({t:"actTrksList", list: recs}); // TODO: split up in multiple transmissions?
+      },
+      //{t:"fetchRec", id:"20230616a"}
+      "fetchRec": function() {
+        // TODO: Decide on what names keys should have.
+        if (event.id=="stop" && fetchRecInterval) {
```

Comment:
@gfwilliams In this if-block I try to stop the sending of Recorder logs short when Gadgetbridge sends the keyword "stop" back. But I haven't gotten it to work yet. Is it that I have the scoping of `fetchRecInterval` wrong I wonder?

Reply:
That all looks fine to me! Can you reproduce it in the IDE just by sending `GB({t:"fetchRec", id:"20230616a"})` and `GB({t:"fetchRec", id:"stop"})`?

I guess one thing I see is if you call `GB({t:"fetchRec", id:"stop"})` when it's not going, it's still going to call `setInterval` which is probably not what you want?

Reply:
Ah, right - thanks!